org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/hover/XbaseDispatchingEObjectTextHover.java

```diff
@@ -84,6 +107,46 @@ public IRegion getHoverRegion(ITextViewer textViewer, int offset) {
 					return Tuples.create(object, original.getSecond());
 			}
 		}
+		
+		Pair<EObject,IRegion> referencedElementInJavaDoc = getReferencedElementInJavaDoc(resource, offset);
+		if(referencedElementInJavaDoc != null) {
+			return referencedElementInJavaDoc;
+		}
+		
 		return original;
 	}
+	
```

Comment:
Looks good. However, this is the same code as in `XbaseHyperLinkHelper#createHyperlinksInJavaDoc`. I would like to see this method relocated to a central utility class or service.

Reply:
Yes, you I agree with you. Refactoring will come in a follow up PR.